From 3e34ddd677aa9c0ddf321890e3bd62a0370102dd Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Thu, 08 May 2025 19:26:22 +0300
Subject: [PATCH] [SW-227605] g2e: support bis-e-compliant preboot

In case reading real fuse or fake fuse from flash fails, then if the card
is a G2 PCIe card (HL288), then assume it is a BIS-Compliant card, which
is a BIS-E-Compliant card. In addition, fix a minor typo in comments,
and not in a separate commit.

Change-Id: I4365b26fa06df5e2beea9d34610386a8a78b9283
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index cbd7eb3..96438a3 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -136,12 +136,12 @@
 				val = &fuse->data[i];
 				*val = ~(*val);
 			}
-			/* HL-288 is Chian only card so BIS D is set by default, unless fake
+		} else if (hl_is_card_type_pcie()) {
+			/* HL-288 is China only card so BIS E is set by default, unless fake
 			 * secutity is enabled
 			 */
-		} else if (hl_is_card_type_pcie()) {
 			fuse->bis_compliant = 1;
-			fuse->bis_d_compliant = 1;
+			fuse->bis_e_compliant = 1;
 		}
 
 		preboot_hl_sec_fuse_cache.is_invalid = false;
@@ -165,10 +165,10 @@
 				return NULL;
 		}
 
-		/* HL-288 is Chian only card so BIS D is set by default */
+		/* HL-288 is China only card so BIS E is set by default */
 		if (hl_is_card_type_pcie()) {
 			fuse->bis_compliant = 1;
-			fuse->bis_d_compliant = 1;
+			fuse->bis_e_compliant = 1;
 		}
 
 		preboot_hl_sec_fuse_cache.is_invalid = false;
@@ -859,22 +859,27 @@
 	return false;
 }
 
+static bool is_bis_compliant_mode(const struct hl_sec_efuse *efuse)
+{
+	if (!efuse) {
+		LOG_ERR("failed to read efuse, forcing bis compliant mode.");
+		return true;
+	}
+
+	return efuse->bis_compliant || efuse->bis_d_compliant || efuse->bis_e_compliant;
+}
+
 bool preboot_check_bis_compliant_mode(void)
 {
 	const struct hl_sec_efuse *efuse;
 
 	if (preboot_security_is_real_efuse_sec_en()) {
 		efuse = preboot_get_hl_sec_real_efuse();
-		if (!efuse) {
-			LOG_ERR("failed to read efuse, forcing bis compliant mode.");
-			return true;
-		}
-
-		return efuse->bis_compliant || efuse->bis_d_compliant;
+		return is_bis_compliant_mode(efuse);
 	}
 
 	efuse = preboot_get_hl_sec_fake_efuse();
-	return efuse->bis_compliant || efuse->bis_d_compliant;
+	return is_bis_compliant_mode(efuse);
 }
 
 bool preboot_is_secondary_image_disabled(void)
